test/distributed/partition_kernels.cpp

```diff
@@ -69,6 +69,16 @@ class Partition : public CommonTestFixture {
             gko::make_array_view(
                 this->exec, dpart->get_num_parts(),
                 const_cast<local_index_type*>(dpart->get_part_sizes())));
+
+        GKO_ASSERT_ARRAY_EQ(
+            gko::make_const_array_view(
+                this->ref, part->get_num_ranges(),
+                part->get_ranges_by_part().get_const_flat_data()),
+            gko::make_const_array_view(
+                this->exec, dpart->get_num_ranges(),
+                dpart->get_ranges_by_part().get_const_flat_data()));
+        GKO_ASSERT_ARRAY_EQ(part->get_ranges_by_part().get_offsets(),
+                            dpart->get_ranges_by_part().get_offsets())
```

Comment:
If you add a `GKO_ASSERT_SEGEMENT_ARRAY_EQ` macro from the assert function in the other file, then you can use that here as well.